File: nk/20220124/L.cpp
#include<bits/stdc++.h>
using namespace std;
double dis(int x,int y){
    return sqrt(x*x+y*y);
}

int main(){
    int t;
    cin>>t;
    while(t--){
        int n;
        cin>>n;
        string s;
        cin>>s;
        double mx=0.0;
        int x=0,y=0;
        for(int i=0;i<n;i++){
            char c=s[i];
            if(c=='U'){
                y++;
            }else if(c=='D'){
                y--;
            }else if(c=='L'){
                x--;
            }else if(c=='R'){
                x++;
            }
            mx = max(dis(x,y),mx);
        }
        cout<<mx<<endl;
    }
    system("pause");
}

File: nk/20220124/L.py


def dis(x,y):
    return (x*x+y*y)**0.5
t=int(input())
for tt in range(t):
    n=int(input())
    s=input()
    x=0
    y=0
    mx=0
    for i in range(n):
        c = s[i]
        if c=='U':
            y+=1
        elif c=='D':
            y-=1
        elif c=='L':
            x-=1
        else: x+=1
        mx=max(float(mx),dis(x,y))
print(mx)

